test/puppetlabs/puppetdb/query_eng_test.clj

```diff
@@ -95,82 +94,3 @@
                         (compile-user-query->sql resources-query ["=" "foo" "bar"]))))
-
-
-(defn fact-names-tests
-  "test that we can modify a query rec. This is a function because the actual
-   test needs to use the with-http-app fixture, which the rest of this ns
-   doesn't require."
-  []
-  (let [version :v4
-        endpoint "/v4/fact-names"
-        facts1 {"domain" "testing.com"
-                "hostname" "foo1"
-                "kernel" "Linux"
-                "operatingsystem" "Debian"
-                "uptime_seconds" "4000"}
-        facts2 {"domain" "testing.com"
-                "hostname" "foo2"
-                "kernel" "Linux"
-                "operatingsystem" "RedHat"
-                "uptime_seconds" "6000"}
-        facts3 {"domain" "testing.com"
-                "hostname" "foo3"
-                "kernel" "Darwin"
-                "operatingsystem" "Darwin"
-                "memorysize" "16.00 GB"}
-        initial-idx @entity-fn-idx]
-
-    (with-transacted-connection fixt/*db*
-      (scf-store/add-certname! "foo1")
-      (scf-store/add-certname! "foo2")
-      (scf-store/add-certname! "foo3")
-      (scf-store/add-facts! {:certname "foo2"
-                             :values facts2
-                             :timestamp (now)
-                             :environment "DEV"
-                             :producer_timestamp (now)})
-      (scf-store/add-facts! {:certname "foo3"
-                             :values facts3
-                             :timestamp (now)
-                             :environment "DEV"
-                             :producer_timestamp (now)})
-      (scf-store/deactivate-node! "foo1")
-      (scf-store/add-facts! {:certname "foo1"
-                             :values  facts1
-                             :timestamp (now)
-                             :environment "DEV"
-                             :producer_timestamp (now)}))
-
-    (let [expected-result ["domain" "hostname" "kernel" "memorysize"
-                           "operatingsystem" "uptime_seconds"]]
-      (testing "fact-names behaves normally"
-        (let [request (get-request endpoint)
-              {:keys [status body]} (fixt/*app* request)
-              result (vec (parse-result body))]
-          (is (= status http/status-ok))
-          (is (= result expected-result))))
-
-      (testing "query rec is modifiable"
-        (assoc-in-idx! [:fact-names :munge] (fn [_ _] identity))
-
-        (assoc-in-idx! [:fact-names :rec :projections "depth"] {:type :integer
-                                                                :queryable? true
-                                                                :field :depth})
-        (let [request (get-request endpoint)
-              {:keys [status body]} (fixt/*app* request)
-              result (vec (parse-result body))]
-          (is (= status http/status-ok))
-          (is (= result (map #(hash-map :name % :depth 0) expected-result)))))
```

Comment:
We also have PE tests for this, so were not completely uninsulated. I went ahead and added the text back either way.